emails/types.py

```diff
@@ -23,18 +23,24 @@
 
 
 class EmailHeaderExceptionOnReadIssue(TypedDict):
+    header: str
+    direction: Literal["in"]
     exception_on_read: str
 
 
 class EmailHeaderExceptionOnWriteIssue(TypedDict):
+    header: str
+    direction: Literal["out"]
     exception_on_write: str
     value: str
 
 
 class EmailHeaderDefectIssue(TypedDict):
+    header: str
+    direction: Literal["in", "out"]
```

Comment:
question (non-blocking): These common fields make me wonder: could the `EmailHeaderExceptionOnReadIssue` and `EmailHeaderExceptionOnWriteIssue` classes extend from this `EmailHeaderDefectIssue` class, and then we might not need the additional `EmailHeaderIssue` type? So we could have `EmailHeaderIssues = list[EmailHeaderDefectIssue]` ?

Reply:
It sounds like you are asking for this:

```python
class EmailHeaderIssue(TypedDict):
    direction: Literal["in", "out"]
    header: str
    defect_count: NotRequired[int]
    parsed_value: NotRequired[str]
    raw_value: NotRequired[str]
    exception_on_read: NotRequired[str]
    value: NotRequired[str]
    exception_on_write: NotRequired[str]
    
EmailHeaderIssues = list[EmailHeaderDefectIssue]
```

That would allow issues like:

```python
issues.append({"direction": "in", "header": "Header"})
issues.append({"direction": "in", "header": "Header", "value": "foo"})
issues.append({"direction": "in", "header": "Header", "defect_count": 3})
issues.append({"direction": "in", "header": "Header", "exception_on_write": "ValueError"})
```

No thanks, I want to type checker to help write correct code.